### PR TITLE
Package rename [OpenTelemetry.Contrib.Extensions.PersistentStorage]

### PR DESCRIPTION
**Pull request by vishweshbankwar**

### CODE REVIEW

**Review by reyang — APPROVED** [review]
LGTM.

File: CODEOWNERS [hunk 1]
```diff
@@ -13,7 +13,7 @@ src/OpenTelemetry.Contrib.Instrumentation.MassTransit/                      @ope
 src/OpenTelemetry.Contrib.Instrumentation.Owin/                             @open-telemetry/dotnet-contrib-approvers @codeblanch
 src/OpenTelemetry.Contrib.Instrumentation.Wcf/                              @open-telemetry/dotnet-contrib-approvers @codeblanch
 src/OpenTelemetry.Contrib.Instrumentation.MySqlData/                        @open-telemetry/dotnet-contrib-approvers @moonheart
-src/OpenTelemetry.Contrib.PersistentStorage/                                @open-telemetry/dotnet-contrib-approvers @vishweshbankwar
+src/OpenTelemetry.Contrib.Extensions.PersistentStorage/                                @open-telemetry/dotnet-contrib-approvers @vishweshbankwar
```
Comment (reyang) [inline CODEOWNERS:16]: nit: update the indentation

File: CODEOWNERS [hunk 2]
```diff
@@ -25,5 +25,5 @@ test/OpenTelemetry.Contrib.Instrumentation.MassTransit.Tests/               @ope
 test/OpenTelemetry.Contrib.Instrumentation.Owin.Tests/                      @open-telemetry/dotnet-contrib-approvers @codeblanch
 test/OpenTelemetry.Contrib.Instrumentation.Wcf.Tests/                       @open-telemetry/dotnet-contrib-approvers @codeblanch
 test/OpenTelemetry.Contrib.Instrumentation.MySqlData.Tests/                 @open-telemetry/dotnet-contrib-approvers @moonheart
-src/OpenTelemetry.Contrib.PersistentStorage.Tests/                          @open-telemetry/dotnet-contrib-approvers @vishweshbankwar
+src/OpenTelemetry.Contrib.Extensions.PersistentStorage.Tests/                          @open-telemetry/dotnet-contrib-approvers @vishweshbankwar
```
Comment (reyang) [inline CODEOWNERS:28]: Similar to https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/181/files#r792171120

File: src/OpenTelemetry.Contrib.Extensions.PersistentStorage/FileStorage.cs [hunk 3]
```diff
@@ -20,7 +20,7 @@
 using System.Linq;
 using System.Timers;
 
-namespace OpenTelemetry.Contrib.PersistentStorage
+namespace OpenTelemetry.Contrib.Extensions.PersistentStorage
```
Comment (reyang) [inline src/OpenTelemetry.Contrib.Extensions.PersistentStorage/FileStorage.cs:23]: Minor - would you consider the C# 10.0 way of saving 4 spaces? 😄

https://github.com/open-telemetry/opentelemetry-dotnet/blob/1150f4a4e43608800b5571da499ed53f237c3904/test/OpenTelemetry.Tests.Stress/Skeleton.cs#L26

**Review by reyang — APPROVED** [review]
LGTM.

File: src/OpenTelemetry.Contrib.Extensions.PersistentStorage/FileStorage.cs (hunk 3 above)
Comment (vishweshbankwar) [reply src/OpenTelemetry.Contrib.Extensions.PersistentStorage/FileStorage.cs]: Done

File: build/Common.props [hunk 4]
```diff
@@ -1,6 +1,6 @@
 <Project>
   <PropertyGroup>
-    <LangVersion>9.0</LangVersion>
+    <LangVersion>10.0</LangVersion>
```
Comment (vishweshbankwar) [inline build/Common.props:3]: @cijothomas - Since this is a small change I included it here. Let me know if I should create a separate pr for this.

Comment (cijothomas) [reply build/Common.props:3]: lets keep it separate as this file affects entire projects, not just persistentstorage. you may do it freely in persistent storage project alone in this PR.

File: src/OpenTelemetry.Contrib.Extensions.PersistentStorage/FileStorage.cs (hunk 3 above)
Comment (vishweshbankwar) [reply src/OpenTelemetry.Contrib.Extensions.PersistentStorage/FileStorage.cs:23]: ran into an issue with this. (CSC : error CS1617: Invalid option '10.0' for /langversion. Use '/langversion:?' to list supported values.) . Will investigate and update this in separate PR.

**Review by cijothomas — APPROVED** [review]

**Pull request merged**